### PR TITLE
Feature/scrum 10 switch to on push strategy

### PR DESCRIPTION
**Pull request by IvanBanifatsyeu**
I'm sorry to have kept you waiting so long

### CODE REVIEW

File: src/app/pokemon-list/pokemon-list.component.ts [hunk 1]
```diff
@@ -35,7 +43,8 @@ export class PokemonListComponent implements OnInit {
   destroyRef = inject(DestroyRef);
   constructor(
     private pokemonService: PokemonService,
-    public dialog: MatDialog
+    public dialog: MatDialog,
```
Comment (MarikaKonturova) [inline src/app/pokemon-list/pokemon-list.component.ts:46]: rewrite constructor to inject

File: src/app/shared/components/search-bar/search-bar.component.ts [hunk 2]
```diff
@@ -12,20 +21,24 @@ import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
...
   imports: [MatIconModule, MatFormFieldModule, ReactiveFormsModule, CommonModule, MatInputModule],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class SearchBarComponent implements OnInit {
   searchControl = new FormControl('');
   @Output() searchEvent = new EventEmitter<{ query: string }>();
   destroyRef = inject(DestroyRef);
-  constructor(private formBuilder: FormBuilder) {}
+  constructor(
+    private formBuilder: FormBuilder,
```
Comment (MarikaKonturova) [inline src/app/shared/components/search-bar/search-bar.component.ts:31]: same as above

File: src/app/shared/components/search-bar/search-bar.component.ts [hunk 3]
```diff
@@ -12,20 +21,24 @@ import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
...
+  ) {}
   ngOnInit() {
     this.searchControl.valueChanges
       .pipe(
         debounceTime(1000),
         tap(searchText => this.searchEvent.emit({ query: searchText ?? '' })),
         takeUntilDestroyed(this.destroyRef)
       )
-      .subscribe();
+      .subscribe(() => this.cdRef.detectChanges());
```
Comment (MarikaKonturova) [inline src/app/shared/components/search-bar/search-bar.component.ts]: check if you really need to have detectChanges here

File: src/app/pokemon-list/pokemon-list.component.ts [hunk 4]
```diff
@@ -46,6 +55,7 @@ export class PokemonListComponent implements OnInit {
       .subscribe(pokemons => {
         this.pokemons = pokemons;
         this.loading = false;
+        this.cdRef.detectChanges();
```
Comment (TKNoProblem) [inline src/app/pokemon-list/pokemon-list.component.ts]: Never use detectChanges if u want to update data in async context

File: src/app/pokemon-list/pokemon-list.component.ts [hunk 5]
```diff
@@ -64,6 +74,7 @@ export class PokemonListComponent implements OnInit {
...
```
Comment (TKNoProblem) [inline src/app/pokemon-list/pokemon-list.component.ts]: Same thing

**Review by TKNoProblem — CHANGES_REQUESTED** [review]

File: src/app/pokemon-list/pokemon-list.component.ts (hunk 4 above)
Comment (IvanBanifatsyeu) [reply src/app/pokemon-list/pokemon-list.component.ts]: 0acf1f06cbd361cd209c3f1c0ca1c1f3cb208437

File: src/app/pokemon-list/pokemon-list.component.ts (hunk 5 above)
Comment (IvanBanifatsyeu) [reply src/app/pokemon-list/pokemon-list.component.ts]: 0acf1f06cbd361cd209c3f1c0ca1c1f3cb208437

File: src/app/pokemon-list/pokemon-list.component.ts (hunk 1 above)
Comment (MarikaKonturova) [reply src/app/pokemon-list/pokemon-list.component.ts:46]: created separate task for that https://pockedex-team.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-6

File: src/app/shared/components/search-bar/search-bar.component.ts (hunk 2 above)
Comment (MarikaKonturova) [reply src/app/shared/components/search-bar/search-bar.component.ts:31]: created separate task for that https://pockedex-team.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-6

File: src/app/shared/components/search-bar/search-bar.component.ts (hunk 3 above)
Comment (MarikaKonturova) [reply src/app/shared/components/search-bar/search-bar.component.ts]: https://github.com/IvanBanifatsyeu/Pokedex-angular19-nx/commit/0acf1f06cbd361cd209c3f1c0ca1c1f3cb208437

**Review by MarikaKonturova — CHANGES_REQUESTED** [review]

**Review by MarikaKonturova — APPROVED** [review]

**Review by TKNoProblem — APPROVED** [review]

**Pull request merged**